unity/Assets/Scripts/ArtGallery/Util/NaiveLighthouseToLighthouseVisibility.cs

```diff
@@ -0,0 +1,279 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Util.Algorithms.Polygon;
+using Util.Geometry;
+using Util.Geometry.Polygon;
+
+namespace ArtGallery
+{
+    public static class NaiveLighthouseToLighthouseVisibility
+    {
+        public static List<Vector2> VisibleVertices(
+            Polygon2D polygon,
+            Vector2 vertex)
+        {
+            List<Vector2> result = new List<Vector2>();
+
+            // calculate new visibility polygon
+            var vision = Visibility.Vision(polygon, vertex);
+
+            if (vision == null)
+            {
+                throw new Exception("Vision polygon cannot be null");
+            }
+
+            //check if the level polygon and the visibility polygon are given 
+            // in the same direction. If not, reverse the vision polygon to
+            // match the direction of the level polygon
+            if (polygon.IsClockwise() != vision.IsClockwise())
+            {
+                vision.Reverse();
+            }
+
+            for (int i = 0; i < polygon.VertexCount; i++)
```

Comment:
Perhaps use a `foreach` loop here?

Reply:
Those were for testing. they are needed with the advanced visibility. I removed them in this branch along with their tests. Apologies